bugbug/model.py

```diff
@@ -41,26 +41,56 @@ def get_feature_names(self):
         return []
 
     def get_important_features(self, cutoff, shap_values):
-        # Calculate the values that represent the fraction of the model output variability attributable
-        # to each feature across the whole dataset.
-        shap_sums = shap_values.sum(0)
-        abs_shap_sums = np.abs(shap_values).sum(0)
-        rel_shap_sums = abs_shap_sums / abs_shap_sums.sum()
-
-        cut_off_value = cutoff * np.amax(rel_shap_sums)
-
-        # Get indices of features that pass the cut off value
-        top_feature_indices = np.where(rel_shap_sums >= cut_off_value)[0]
-        # Get the importance values of the top features from their indices
-        top_features = np.take(rel_shap_sums, top_feature_indices)
-        # Gets the sign of the importance from shap_sums as boolean
-        is_positive = (np.take(shap_sums, top_feature_indices)) >= 0
-        # Stack the importance, indices and shap_sums in a 2D array
-        top_features = np.column_stack((top_features, top_feature_indices, is_positive))
-        # Sort the array (in decreasing order of importance values)
-        top_features = top_features[top_features[:, 0].argsort()][::-1]
-
-        return top_features
+        # shap_values is a list for models with vector output
+        if isinstance(shap_values, list):
+            shap_values = np.array(shap_values)
+            class_shap_values = []
+
+            for matrix in shap_values:
+                shap_sum = (np.abs(matrix)).sum(0)
+                class_shap_values.append(shap_sum)
+
+            # Importance of features (sum of all classes)
+            feature_shap_vals = np.array(class_shap_values).sum(0)
+            sorted_feature_indices = np.argsort(feature_shap_vals)[::-1]
+            # features: matrix of [features x classes] sorted by importance of features
+            features = (np.array(class_shap_values).T)[sorted_feature_indices]
+
+            cut_off_value = cutoff * np.amax(feature_shap_vals)
+            # Gets the number of features that pass the cut off value
+            cut_off_len = len(np.where(feature_shap_vals >= cut_off_value)[0])
+
+            # Stack the original indices and feature importance along with the features
+            # above the cut off length
+            top_features = np.column_stack((
+                sorted_feature_indices[:cut_off_len],
+                feature_shap_vals[sorted_feature_indices][:cut_off_len],
+                features[:cut_off_len]
+            ))
+
+            return top_features
+
+        else:
```

Comment:
Couldn't the one-level case be handled by doing something like `shap_values = [shap_values]`, so that we share the same code for single-class and multiclass problems?

Reply:
- Single class importances are sorted according to relative shap sums, multi-class uses just shap sums

 - In single class case, we can effectively determine the sign of the importance value whereas in multi-class we use sum of absolute values

So I thought it would be better to leave single-class as it is, but I can merge them if you want.

Reply:
Couldn't we determine the sign in the multiclass case?

We can sort by absolute value to have the most important features to show, but then we can show the sign.

Reply:
|  | Feature 1 (most important between all classes) | Feature 2 (second most important between all classes) |
| ------------- |:-------------:|:-------------:|
|  **Class1** | +1,3 | -1,2 |
| **Class2** | +0,01 | +7,9 |
|  **Class3** | +99,3 | -11,2 |